2.15.3 test_compile_to_code_custom_format_with_refs fails

Hi! When packaging this project for Arch Linux I ran into the test `test_compile_to_code_custom_format_with_refs` failing for me:

```
=================================== FAILURES ===================================
_________________ test_compile_to_code_custom_format_with_refs _________________

    def test_compile_to_code_custom_format_with_refs():
        schema = {
            'type': 'object',
            'properties': {
                'a': {'$ref': '#/definitions/a'}
            },
            'definitions': {
                'a': {
                    '$id': '#/definitions/a',
                    'type': 'array',
                    'items': {'type': 'string', 'format': 'my-format'}
                }
            }
        }
        formats = {'my-format': str.isidentifier}
        code = compile_to_code(schema, formats=formats)
        with open('temp/schema_4.py', 'w') as f:
            f.write(code)
>       from temp.schema_4 import validate
E       ModuleNotFoundError: No module named 'temp.schema_4'

tests/test_compile_to_code.py:120: ModuleNotFoundError
=============================== warnings summary ===============================
tests/test_compile_to_code.py:8
  /build/python-fastjsonschema/src/python-fastjsonschema/tests/test_compile_to_code.py:8: PytestDeprecationWarning: @pytest.yield_fixture is deprecated.
  Use @pytest.fixture instead; they are the same.
    @pytest.yield_fixture(autouse=True)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_compile_to_code.py::test_compile_to_code_custom_format_with_refs
= 1 failed, 2024 passed, 12 deselected, 8 xfailed, 8 xpassed, 1 warning in 6.68s =
```

Full build log: 
[python-fastjsonschema-2.15.3-build.log](https://github.com/horejsek/python-fastjsonschema/files/8056737/python-fastjsonschema-2.15.3-build.log)

I am running pytest without benchmarks (i.e. `pytest -v -m "not benchmark"`) and I do not have any further pytest related packages installed during test.
Is this maybe related to some feature of pytest-cache?

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,0 +1,31 @@
+name: tests
+
+on: [push, pull_request, workflow_dispatch]
+
+concurrency:
+  group: >-
+    ${{ github.workflow }}-
+    ${{ github.ref_type }}-
+    ${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        python:
+        - 3.7
+        - "3.10"
+        platform:
+        - ubuntu-latest
+        - macos-latest
+        - windows-latest
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Run tests
+        run: pipx run tox -e py -- -vvv

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -14,8 +14,8 @@ jobs:
     strategy:
       matrix:
         python:
-        - 3.7
-        - "3.10"
+        - 3.7  # oldest Python supported by PSF
+        - "3.10"  # newest Python that is stable
         platform:
         - ubuntu-latest
         - macos-latest

--- a/tests/test_compile_to_code.py
+++ b/tests/test_compile_to_code.py
@@ -116,7 +116,7 @@ def test_compile_to_code_custom_format_with_refs(tmp_path, monkeypatch):
     formats = {'my-format': str.isidentifier}
     code = compile_to_code(schema, formats=formats)
 
-    (tmp_path / "schema_4.py").write_text(code)
+    (tmp_path / "schema_4.py").write_text(code, encoding="utf-8")
     with monkeypatch.context() as m:
         m.syspath_prepend(tmp_path)
         from schema_4 import validate

--- a/tests/test_compile_to_code.py
+++ b/tests/test_compile_to_code.py
@@ -99,7 +99,7 @@ def test_compile_to_code_custom_format():
     assert exc.value.message == "data must be my-format"
 
 
-def test_compile_to_code_custom_format_with_refs():
+def test_compile_to_code_custom_format_with_refs(tmp_path, monkeypatch):
     schema = {
         'type': 'object',
         'properties': {
@@ -115,9 +115,12 @@ def test_compile_to_code_custom_format_with_refs():
     }
     formats = {'my-format': str.isidentifier}
     code = compile_to_code(schema, formats=formats)
-    with open('temp/schema_4.py', 'w') as f:
-        f.write(code)
-    from temp.schema_4 import validate
+
+    (tmp_path / "schema_4.py").write_text(code)
+    with monkeypatch.context() as m:
+        m.syspath_prepend(tmp_path)
+        from schema_4 import validate
+
     assert validate({"a": ["identifier"]}, formats) is not None
     with pytest.raises(JsonSchemaValueException) as exc:
         validate({"a": ["identifier", "not-valid"]}, formats)

--- a/tests/test_examples.py
+++ b/tests/test_examples.py
@@ -30,9 +30,9 @@ EXAMPLE_GROUPS = [x.name for x in EXAMPLES.glob("*/")]
 def test_validate(example_group):
     example_dir = EXAMPLES / example_group
     schema = next(example_dir.glob("*.schema.json"))
-    validator = compile(json.loads(schema.read_text()))
+    validator = compile(json.loads(schema.read_text(encoding="utf-8")))
     examples = (
-        (e, e.with_suffix(".error"), json.loads(e.read_text()))
+        (e, e.with_suffix(".error"), json.loads(e.read_text(encoding="utf-8")))
         for e in example_dir.glob("*.json")
         if not str(e).endswith(".schema.json")
     )
@@ -40,6 +40,6 @@ def test_validate(example_group):
         if err_file.exists():
             with pytest.raises(JsonSchemaException) as exc_info:
                 validator(example)
-            assert err_file.read_text().strip() in str(exc_info.value).strip()
+            assert err_file.read_text(encoding="utf-8").strip() in str(exc_info.value).strip()
         else:
             validator(example)  # valid

--- a/tox.ini
+++ b/tox.ini
@@ -10,7 +10,7 @@ envlist = py{34,35,36,37},lint
 deps =
     pytest
 commands =
-    pytest -m "not benchmark"
+    pytest -m "not benchmark" {posargs}
 
 [testenv:lint]
 deps =

